### PR TITLE
Adding the Hyper file defragmentation script (python)

### PR DESCRIPTION
**Pull request by d45**
@tsjoblad @jonas-eckhardt 

This is the python script we reference from the Hyper API Help topic about optimizing Hyper file storage.

### CODE REVIEW

File: Python/defragment_data_of_existing_hyper_file.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import argparse
+from pathlib import PurePath
+from tableauhyperapi import HyperProcess, Telemetry, Connection, TableDefinition, TableName
+
```
Comment (jonas-eckhardt) [inline Python/defragment_data_of_existing_hyper_file.py:4]: Can we please add a general comment to this script what it does?

File: Python/defragment_data_of_existing_hyper_file.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+if __name__ == "__main__":
+    argparser = argparse.ArgumentParser(description="This tool rewrites a given Hyper file in an optimized, dense format.")
+    argparser.add_argument("input_hyper_file_path", type=PurePath, help="The input Hyper file path that will be rewritten")
```
Comment (jonas-eckhardt) [inline Python/defragment_data_of_existing_hyper_file.py:7]: . at the end of the sentence

File: Python/defragment_data_of_existing_hyper_file.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    argparser.add_argument("--output-hyper-file-path", "-o", type=PurePath, help="The output Hyper file path for the rewritten output Hyper file")
```
Comment (jonas-eckhardt) [inline Python/defragment_data_of_existing_hyper_file.py:8]: . at the end of the sentence

File: Python/defragment_data_of_existing_hyper_file.py (hunk 2 above)
Comment (jonas-eckhardt) [inline Python/defragment_data_of_existing_hyper_file.py:7]: we should make the input file required (Required=True)

File: Python/defragment_data_of_existing_hyper_file.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    if not args.output_hyper_file_path:
+        args.output_hyper_file_path = args.input_hyper_file_path.parent / (args.input_hyper_file_path.stem + f".new.hyper")
+
+    with HyperProcess(telemetry=Telemetry.SEND_USAGE_DATA_TO_TABLEAU) as hyper:
+        with Connection(endpoint=hyper.endpoint) as connection:
+            # Create the output Hyper file or overwrite it
+            catalog = connection.catalog
+            catalog.drop_database_if_exists(args.output_hyper_file_path)
+            catalog.create_database(args.output_hyper_file_path)
+            catalog.attach_database(args.output_hyper_file_path, alias="output_database")
```
Comment (jonas-eckhardt) [inline Python/defragment_data_of_existing_hyper_file.py:19]: Add a comment along the lines "Connect to the input and output databases"

File: README.md [hunk 5]
```diff
@@ -47,7 +47,8 @@ There are currently eight samples that are available in each language supported:
 * __insert_data_using_expressions__
   * This example shows how you can use SQL expressions in Hyper API Inserter to transform or compute data on the fly during data insertion
 
-
+* __defragment_data_of_existing_hyper_file__
```
Comment (jonas-eckhardt) [inline README.md:50]: Can we please add the "Community supported" tag?

File: Python/defragment_data_of_existing_hyper_file.py (hunk 1 above)
Comment (d45) [reply Python/defragment_data_of_existing_hyper_file.py:4]: Added: 
 An example of how you can optimize the file storage of an existing `.hyper` file by copying 
 all of the tables and data into a new file. This reduces file fragmentation.

File: Python/defragment_data_of_existing_hyper_file.py (hunk 3 above)
Comment (d45) [reply Python/defragment_data_of_existing_hyper_file.py:8]: done

File: README.md (hunk 5 above)
Comment (d45) [reply README.md:50]: Just this sample? Or all samples (at the repo level)?

Comment (d45) [reply README.md:50]: I added it to just this sample. @tsjoblad  @jonas-eckhardt  I'm not sure how you were planning to split the community supported samples from the Tableau supported ones. I just threw the label in here for this one.

Comment (jonas-eckhardt) [reply README.md:50]: Please add it at the sample level. We will put each of them in individual folders, each with its own readme + support level.

@tsjoblad, right?

**Comment by tsjoblad** [comment]
I'm closing this pull request as a lot has changed in the repo. I took the .py file from the pr and placed it into my dev branch.

**Pull request closed without merging**